File: largest_files.py
import os

# Folder to scan
folder = r"C:\Users\Baha\Desktop\Developpement full stack\Lab 6"  # Change if needed

# Get all files with their sizes
files = []
for filename in os.listdir(folder):
    file_path = os.path.join(folder, filename)
    if os.path.isfile(file_path):
        size = os.path.getsize(file_path)  # size in bytes
        files.append((filename, size))

# Sort files by size descending
files.sort(key=lambda x: x[1], reverse=True)

# Take top 5
top5 = files[:5]

print("[OK] 5 plus gros fichiers dans le dossier:")
for name, size in top5:
    print(f"{name} - {size / 1024:.2f} KB")

File: Go largest_files/largest_files.go
package main

import (
	"fmt"
	"io/ioutil"
	"log"
	"sort"
)

type FileInfo struct {
	Name string
	Size int64
}

func main() {
	folder := `C:\Users\Baha\Desktop\Developpement full stack\Lab 6` // Change if needed

	files, err := ioutil.ReadDir(folder)
	if err != nil {
		log.Fatal(err)
	}

	var fileInfos []FileInfo
	for _, f := range files {
		if !f.IsDir() {
			fileInfos = append(fileInfos, FileInfo{Name: f.Name(), Size: f.Size()})
		}
	}

	// Sort by size descending
	sort.Slice(fileInfos, func(i, j int) bool {
		return fileInfos[i].Size > fileInfos[j].Size
	})

	fmt.Println("[OK] 5 plus gros fichiers dans le dossier:")
	for i, f := range fileInfos {
		if i >= 5 {
			break
		}
		fmt.Printf("%s - %.2f KB\n", f.Name, float64(f.Size)/1024)
	}
}
